### ISSUE TITLE
Exercise 1 in Node Classification notebook

### ISSUE THREAD

**Issue by vinhtq115** [rusty1s/pytorch_geometric#1586]
## ❓ Questions & Help

I'm trying to do the exercise 1 in [Node Classification notebook](https://colab.research.google.com/drive/14OvFnAXggxB8vM4e8vSURUp1TaKnovzX?usp=sharing) and I'm not sure whether my code is correct:

```
model = GCN(hidden_channels=16)
optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
criterion = torch.nn.CrossEntropyLoss()

best_val = -1.0
best_model = GCN(hidden_channels=16)

def train():
    model.train()
    out = model(data.x, data.edge_index)  # Perform a single forward pass.
    loss = criterion(out[data.train_mask], data.y[data.train_mask])  # Compute the loss solely based on the training nodes.
    loss.backward()  # Derive gradients.
    optimizer.step()  # Update parameters based on gradients.
    optimizer.zero_grad()  # Clear gradients.
    return loss

def test():
    best_model.eval()
    out = best_model(data.x, data.edge_index)
    pred = out.argmax(dim=1)  # Use the class with highest probability.
    test_correct = pred[data.test_mask] == data.y[data.test_mask]  # Check against ground-truth labels.
    test_acc = int(test_correct.sum()) / int(data.test_mask.sum())  # Derive ratio of correct predictions.
    return test_acc

def val():
    model.eval()
    out = model(data.x, data.edge_index)
    pred = out.argmax(dim=1)  # Use the class with highest probability.
    val_correct = pred[data.val_mask] == data.y[data.val_mask]  # Check against ground-truth labels.
    val_acc = int(val_correct.sum()) / int(data.val_mask.sum())  # Derive ratio of correct predictions.
    return val_acc

for epoch in range(1, 201):
    loss = train()
    if best_val < val():
        best_val = val()
        best_model.load_state_dict(model.state_dict())
    
    _val = val()
    print(f'Epoch: {epoch:03d}, Loss: {loss:.4f}, Val: {_val:.4f}, Best Val: {best_val:.4f}')

test_acc = test()
print(f'Test Accuracy: {test_acc:.4f}')
```

After running the code, I got `Test Accuracy: 0.7940`, which is a bit lower than `Test Accuracy: 0.7970` when running without validation. Is this normal for this exercise? Also, if you have done the exercise, could I ask what is your final result?

**Comment by rusty1s** [comment]
Hi and thanks for this issue :) It's great to see you are already looking at the brand-new colab notebooks.

Your code is absolutely correct, which made me aware that I was missing a crucial step to achieve better performance for `GCN` on `Planetoid` datasets: I have forgotten to normalize input node features via:
```python
dataset = Planetoid(root='data/Planetoid', name='Cora', transform=NormalizeFeatures())
```
This is now fixed, and leads to 81.4% test accuracy without validation, and 82% test accuracy with validation.
